applications/plugins/SofaSphFluid/src/SofaSphFluid/ParticleSink.inl

```diff
@@ -62,14 +65,30 @@ void ParticleSink<DataTypes>::init()
     this->core::behavior::ProjectiveConstraintSet<DataTypes>::init();
     if (!this->mstate) return;
 
-    msg_info() << "Normal=" << d_planeNormal.getValue() << " d0=" << d_planeD0.getValue() << " d1=" << d_planeD1.getValue();
+    // If no topology set, check if one is valid in the node
+    if (l_topology.empty())
+    {
+        l_topology.set(this->getContext()->getMeshTopologyLink());
+    }
 
-    sofa::core::topology::BaseMeshTopology* _topology;
-    _topology = this->getContext()->getMeshTopology();
+    sofa::core::topology::BaseMeshTopology* _topology = l_topology.get();
 
-    // Initialize functions and parameters for topology data and handler
-    d_fixed.createTopologyHandler(_topology);
-    d_fixed.registerTopologicalData();
+    if (_topology) // If topology is found, will check if it is dynamic. Otherwise only mechanical Object should be used.
+    {
+        _topology->getContext()->get(m_topoModifier);
+        if (!m_topoModifier)
+        {
+            sofa::core::objectmodel::BaseObject::d_componentState.setValue(sofa::core::objectmodel::ComponentState::Invalid);
+            msg_error() << "Topology has been found in at component: " << _topology->getName() << " but this topology is not dynamyc. ParticleSink will not be able to remove points.";
```

Comment:
- Should it be a warning instead ?
- and typo (_dynamic_)

Reply:
component is set to invalid so no it should be an error.

Reply:
Then Components/collision.SpatialGridPointModel.scn and Components/visualmodel.PointSplatModel.scn should be fixed (raising this msg_error)

Reply:
this means those scenes depends on the plugin?